src/ray/raylet/worker_pool.cc

```diff
@@ -91,11 +90,13 @@ void WorkerPool::RegisterWorker(std::shared_ptr<Worker> worker) {
   auto pid = worker->Pid();
   RAY_LOG(DEBUG) << "Registering worker with pid " << pid;
   registered_workers_.push_back(std::move(worker));
+
   auto it = starting_worker_processes_.find(pid);
-  RAY_CHECK(it != starting_worker_processes_.end());
-  it->second--;
-  if (it->second == 0) {
-    starting_worker_processes_.erase(it);
+  if (it != starting_worker_processes_.end()) {
```

Comment:
why would this happen?

Reply:
@raulchen  I have no better idea about it.

When the worker which is started by the java-runmanager posts a `register-msg` to node manager,  the `starting_worker_processes_ ` have no information about it. So  if we `RAYCHECK(it != starting_worker_processes_.end())` , it will not be passed.

Do you have a better idea?